base_tier_validation_board/models/tier_review.py

```diff
@@ -0,0 +1,161 @@
+    )
+
+    @api.depends("res_id", "model")
+    def _compute_related_model_instance(self):
+        for record in self:
+            ref = False
+            if record.res_id:
+                ref = f"{record.model},{record.res_id}"
+            record.related_model_instance = ref
+
+    @api.model
+    def _selection_related_model_instance(self):
+        models = (
+            self.env["tier.definition"]
+            .sudo()
+            .search_fetch([], ["model_id"])
+            .mapped("model_id")
+        )
+        models.fetch(["model", "name"])
+        return [(model.model, model.name) for model in models]
+
+    @api.depends("model", "res_id")
+    def _compute_res_name(self):
+        for record in self:
+            if record.res_id and record.model:
+                record.res_name = (
+                    self.env[record.model]
+                    .search_fetch([("id", "=", record.res_id)], ["display_name"])
+                    .display_name
+                )
+            else:
+                record.res_name = False
+
+    def open_origin(self):
+        self.ensure_one()
+        vid = self.env[self.model].browse(self.res_id).get_formview_id()
+        response = {
+            "type": "ir.actions.act_window",
+            "res_model": self.model,
+            "view_mode": "form",
+            "res_id": self.res_id,
+            "target": "current",
+            "views": [(vid, "form")],
+        }
+        return response
+
+    @api.model
+    def _search(self, domain, offset=0, limit=None, order=None):
+        # Rules do not apply to administrator
+        if self.env.is_superuser():
+            return super()._search(domain, offset, limit, order)
+        # Perform a super with count as False, to have the ids, not a counter
+        query = super()._search(domain, offset, limit, order)
+
+        ids = self.browse(query).ids
+        if not ids:
+            return query
+
+        super(TierReview, self.with_user(self._uid)).check_access("read")
+
+        self.flush_model(["model", "res_id"])
+        reviews_to_check = []
+        for sub_ids in self._cr.split_for_in_conditions(ids):
+            self._cr.execute(
+                SQL(
+                    """
+                SELECT DISTINCT review.id, review.model, review.res_id
+                FROM %(table)s review
+                WHERE review.id = ANY (%(ids)s) AND review.res_id != 0""",
+                    table=SQL.identifier(self._table),
+                    ids=list(sub_ids),
+                )
+            )
+            reviews_to_check += self._cr.dictfetchall()
+
+        review_to_documents = {}
+        for review in reviews_to_check:
+            review_to_documents.setdefault(review["model"], set()).add(review["res_id"])
+
+        allowed_ids = set()
+        for doc_model, doc_ids in review_to_documents.items():
+            doc_operation = "read"
+            DocumentModel = self.env[doc_model].with_user(self._uid)
```

Comment:
```suggestion
            DocumentModel = self.env[doc_model]
```
Because there was no argument `access_rights_uid` as previous version.